main.go

```diff
@@ -64,13 +65,23 @@ func main() {
 		return
 	}
 
+	// default to config port if the sftp was not passed.
+	if !isFlagPassed("port") {
+		confPort, err := jsonparser.GetInt(config, "sftp", "port")
+		if err != nil {
+			logger.Get().Debugw("could not find sftp port, falling back to \"2022\"", zap.Error(err))
```

Comment:
Where is the fallback happening here? It looks like if there is an error it doesn't actually pop out of this process and will continue trying to use a port from the config?

Also below this, you can zap.Int I think, rather than the `strconv`?

Reply:
If there was a flag passed it skips this all together.

If the flag was not passed it reads the config and uses the config value.

If there is no config value it will error and stick to the default flag value of 2022.